### PR TITLE
SIMD instruction speed up the storage layer

### LINKED ISSUE #6088: [Performance] Use SMID instruction to speed up the page decode of PlainPage and DictPage
## Motivation
Now, When we run ssb test for doris. See the CPU Perf find：
image

There is plenty of CPU compute in `page decode of PlainPage and DictPage`

try to see the detail, we find there are many of mem allocate in dispose the `BitUtil::RoundUpToPowerOf2`
image

## Implementation

Obvious, we can use the SMID to speed up the function `BitUtil::RoundUpToPowerOf2`

After use SSE to speed up the function, the perf show CPU cost:

image

|   |  no vectorized  | vectorized |
|  ---- |  ----  | ----  |
| DictPage| 23.42% |  14.82% |
| PlainPage| 23.38%  |  11.93% |

### 3. More Test In SSB

image

We can find q4，q5，q6，q8，q9，q11 improve about 20%

### PR DESCRIPTION
**Pull request by HappenLee**
## Proposed changes

RT

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Code refactor (Modify the code structure, format the code, etc...)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have created an issue on (Fix #6088) and described the bug/feature there in detail
- [x] Compiling and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] If these changes need document changes, I have updated the document
- [x] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/util/bit_util.h [hunk 1]
```diff
@@ -300,6 +300,11 @@ class BitUtil {
         return (value + (factor - 1)) & ~(factor - 1);
     }
 
+    // speed up function compute for SIMD
+    static inline size_t RoundUpToPowerOf2Int32(size_t value, size_t factor) {
+        return (value + (factor - 1)) & ~(factor - 1);
```
Comment (wangbo) [inline be/src/util/bit_util.h:305]: What's the keypoint for auto SIMD here?
Just remove ```DCHECK``` is ok?

Comment (HappenLee) [reply be/src/util/bit_util.h:305]: 1. keypoint is `inline`.
GCC can do auto SIMD in this `for loop`:
`        auto mem_size = 0;
        for (int i = 0; i < max_fetch; ++i) {
            mem_len[i] = (value + (8 - 1)) & ~(8 - 1);
            mem_size += mem_len[i];
        }
`

2. I will Add DCHECK here

File: be/src/olap/rowset/segment_v2/binary_dict_page.cpp [hunk 2]
```diff
@@ -248,21 +249,37 @@ Status BinaryDictPageDecoder::next_batch(size_t* n, ColumnBlockView* dst) {
     ColumnBlock column_block(_batch.get(), dst->column_block()->pool());
     ColumnBlockView tmp_block_view(&column_block);
     RETURN_IF_ERROR(_data_page_decoder->next_batch(n, &tmp_block_view));
-    for (int i = 0; i < *n; ++i) {
+    const auto len = *n;
+
+    std::vector<size_t> mem_len(len);
```
Comment (wangbo) [inline be/src/olap/rowset/segment_v2/binary_dict_page.cpp:254]: How about ```size_t mem_len[len]``` here?
it's more lightweight.

File: be/src/olap/rowset/segment_v2/binary_plain_page.h [hunk 3]
```diff
@@ -193,18 +194,33 @@ class BinaryPlainPageDecoder : public PageDecoder {
...
+
+        // allocate a batch of memory and do memcpy
+        out = reinterpret_cast<Slice*>(dst->data());
+        char* destination = (char*)dst->column_block()->pool()->allocate(mem_size);
+        if (destination == nullptr) {
+            return Status::MemoryAllocFailed(
+                strings::Substitute("memory allocate failed, size:$0", mem_size));
+        }
+        for (int i = 0; i < max_fetch; ++i) {
+            out->relocate(destination);
```
Comment (wangbo) [inline be/src/olap/rowset/segment_v2/binary_plain_page.h:221]: Here still need to memcpy ```max_fetch``` times;
If we use CK's ```ColumnString``` to replace current's ColumnBlock, just memcpy once here.
We can do it later;

**Review by wangbo — APPROVED** [review]
LGTM

**Pull request merged**